controllers/scripts/common-env.sh

```diff
@@ -103,8 +103,70 @@ export MAPPED_PORT="$POD_PORT"
 export MAPPED_TLSPORT="$POD_TLSPORT"
 {{- end}}
 
+# parseCustomNetworkIP function parses the network IPs for the given list of network names
+# It parses network status information from pod annotations key `k8s.v1.cni.cncf.io/network-status` which is added by CNI
+parseCustomNetworkIP() {
+  local networks=$1
+  networks="$(echo "${networks}" | sed 's/\[//g'| sed 's/\]//g')"
+
+  DATA="$(curl --cacert $CA_CERT -H "Authorization: Bearer $TOKEN" "$KUBE_API_SERVER/api/v1/namespaces/$NAMESPACE/pods/$MY_POD_NAME")"
+
+  NETWORKIPS="$(echo $DATA | python3 -c "import sys, json
+data = json.load(sys.stdin);
+networks = '${networks}';
+networks = networks.strip().split(' ')
+def getNetworkIP(data, networks):
+    annotations = data['metadata']['annotations']
+    key = 'k8s.v1.cni.cncf.io/network-status'
+    if key in annotations:
+      netIPs = []
+      for net in json.loads(annotations[key]):
+        if net['name'] in networks:
+          if 'ips' in net:
+            if len(net['ips']) == 0:
+              raise ValueError(\"ips list empty for network {} in pod annotations key k8s.v1.cni.cncf.io/network-status\".format(net['name']))
+            netIPs.extend(net['ips'])
+          else:
+            raise KeyError(\"ips key missing for network {} in pod annotations key k8s.v1.cni.cncf.io/network-status\".format(net['name']))
+
+      if len(netIPs) == 0:
+        raise ValueError(\"networks {} not found in pod annotations key k8s.v1.cni.cncf.io/network-status\".format(networks))
+      return ' '.join(netIPs)
+print(getNetworkIP(data, networks))")"
+}
+
+{{- if eq .NetworkPolicy.AccessType "customInterface"}}
+parseCustomNetworkIP "{{ .NetworkPolicy.CustomAccessNetworkNames}}"
```

Comment:
any reason for a space between {{ and .Network?

Reply:
Remove space to keep it consistent